Server/mods/deathmatch/logic/lua/CLuaModule.cpp

```diff
@@ -297,59 +292,56 @@ bool CLuaModule::RegisterFunction(lua_State* luaVM, const char* szFunctionName,
     return true;
 }
 
-bool CLuaModule::GetResourceName(lua_State* luaVM, std::string& strName)
+std::string CLuaModule::GetResourceName(lua_State* luaVM)
 {
-    if (luaVM)
-    {
-        CLuaMain* pLuaMain = m_pLuaModuleManager->GetLuaManager()->GetVirtualMachine(luaVM);
-        if (pLuaMain)
-        {
-            CResource* pResource = pLuaMain->GetResource();
-            if (pResource)
-            {
-                strName = pResource->GetName();
-                return true;
-            }
-        }
-    }
-    return false;
+    if (!luaVM)
+        return "";
+
+    CLuaMain* pLuaMain = m_pLuaModuleManager->GetLuaManager()->GetVirtualMachine(luaVM);
+    if (!pLuaMain)
+        return "";
+
+    CResource* pResource = pLuaMain->GetResource();
+    if (!pResource)
+        return "";
+
+    return pResource->GetName();
 }
 
 CChecksum CLuaModule::GetResourceMetaChecksum(lua_State* luaVM)
 {
-    if (luaVM)
-    {
-        CLuaMain* pLuaMain = m_pLuaModuleManager->GetLuaManager()->GetVirtualMachine(luaVM);
-        if (pLuaMain)
-        {
-            CResource* pResource = pLuaMain->GetResource();
-            if (pResource)
-            {
-                return pResource->GetLastMetaChecksum();
-            }
-        }
-    }
-    return CChecksum();
+    if (!luaVM)
+        return CChecksum();
+
+    CLuaMain* pLuaMain = m_pLuaModuleManager->GetLuaManager()->GetVirtualMachine(luaVM);
+    if (!pLuaMain)
+        return CChecksum();
+
+    CResource* pResource = pLuaMain->GetResource();
+    if (!pResource)
+        return CChecksum();
+
+    return pResource ? pResource->GetLastMetaChecksum() : CChecksum();
 }
 
 CChecksum CLuaModule::GetResourceFileChecksum(lua_State* luaVM, const char* szFile)
 {
-    if (luaVM)
+    if (!luaVM)
+        return CChecksum();
+
+    CLuaMain* pLuaMain = m_pLuaModuleManager->GetLuaManager()->GetVirtualMachine(luaVM);
+    if (!pLuaMain)
+        return CChecksum();
+
+    CResource* pResource = pLuaMain->GetResource();
+    if (!pResource)
+        return CChecksum();
+
+    list<CResourceFile*>::iterator iter = pResource->IterBegin();
+    for (; iter != pResource->IterEnd(); ++iter)
     {
-        CLuaMain* pLuaMain = m_pLuaModuleManager->GetLuaManager()->GetVirtualMachine(luaVM);
-        if (pLuaMain)
-        {
-            CResource* pResource = pLuaMain->GetResource();
-            if (pResource)
-            {
-                list<CResourceFile*>::iterator iter = pResource->IterBegin();
-                for (; iter != pResource->IterEnd(); ++iter)
-                {
-                    if (strcmp((*iter)->GetName(), szFile) == 0)
-                        return (*iter)->GetLastChecksum();
-                }
-            }
-        }
+        if (!strcmp((*iter)->GetName(), szFile))
+            return (*iter)->GetLastChecksum();
```

Comment:
You could use a range for here too, no?

Reply:
No, you cant. Range loop work only on default `begin` and `end` functions